### DISCUSSION TITLE
Adding a public method to retrieve input from the request as an array

### DISCUSSION THREAD

**Discussion by lkeio** [laravel/framework#51738] [Ideas]
In the `Illuminate\Http\Concerns\InteractsWithInput` trait, we can see some helpers to retrieve data already typed like:

```php
    /**
     * Retrieve input from the request as a Stringable instance.
     *
     * @param  string  $key
     * @param  mixed  $default
     * @return \Illuminate\Support\Stringable
     */
    public function string($key, $default = null)
    {
        return str($this->input($key, $default));
    }
```

I was wondering if it could be a good idea to add a method for an array:
Something like this:

```php
    /**
     * Retrieve input from the request as a Stringable instance.
     *
     * @param  string  $key
     * @param  mixed  $default
     * @return array
     */
    public function array(string $key, array $default = []): array
    {
        $value = $this->input($key, $default);

        return is_array($value) ? $value : $default;
    }
```

Let me know, if it makes sense to have a feature like this.

**Comment by ghost** [comment]
?param[]=1
\request()->get('param', []); // is array from php


or

?param=1
(array)\request()->get('param', '');


The logic that if not array it should return default I see it as validation... so better rename the function to arrayOr

**Reply by lkeio** [reply]
I don't get what you saying, it's not a validation it's a helper, you can see similar helpers in here: `Illuminate\Http\Concerns\InteractsWithInput::string`, `Illuminate\Http\Concerns\InteractsWithInput::enum` etc...

The idea is in your controller or whatever, you can use this new helper to fetch data already validated as an array:

i'll explain in detail how to use it:
**Validator:**
```php
use Illuminate\Support\Facades\Validator;

Validator::make($input, [
    'user' => 'array:name,username',
]);
```

**Usage:**
```php
$request->array('user');
```

i hope i explained better now.

**Reply by ghost** [reply]
I am saying that if you want analog function for array, just like ->string function  is, that is just a cast without if in it, then you should rename your function arrayOr to make it obvious that it has a condition in it.

**Reply by ghost** [reply]
![image](https://github.com/laravel/framework/assets/82870454/01cebc7e-ade8-426f-bd4b-1ebcd167dc62)
I don't know who created that ->string function but it's poorly documented. It throws exceptions... for this case:

?param[]=1 //(you can have this also as a form param not just as query param and it is transformed into an array in php.)
\request()->string('param', '');

Oh how I wish I could NOT load this kind of happy flow code into my laravel projects... Useless overhead..

![image](https://github.com/laravel/framework/assets/82870454/d75a5967-c5db-4b4d-8c78-d9fcc4e7fce2)


![image](https://github.com/laravel/framework/assets/82870454/c250e311-524e-4dae-ab5e-5884ce3dae7c)

Stringable class:
![image](https://github.com/laravel/framework/assets/82870454/89342aa7-cae2-4abc-a37c-8701b6847473)

**Reply by ghost** [reply]
Also makes no sense in validating the input as array and then calling ->array function. \request()->get('param', []); will be an array in that case.

**Comment by henzeb** [comment]
You have ->json(). That returns an array.

**Reply by ghost** [reply]
He is not talking about the whole request, but just a key from it.

**Reply by henzeb** [reply]
That's what you can do with json...

https://laravel.com/docs/11.x/requests#retrieving-json-input-values
